app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -259,6 +260,12 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
 
         ProcessLifecycleOwner.get().lifecycle.addObserver(TelemetryLifecycleObserver(components.core.store))
 
+        UsageRecorder(components.settings).let {
```

Comment:
Instead of creating the `UsageReporter` instance for every `Application#onCreate`, maybe make the `shouldRegister` a static function and wrap the instantiation and register instead that if-check?